ksp_plugin/interface_planetarium.cpp

```diff
@@ -107,117 +122,147 @@ void __cdecl principia__PlanetariumDelete(
   return m.Return();
 }
 
-Iterator* __cdecl principia__PlanetariumPlotFlightPlanSegment(
+void __cdecl principia__PlanetariumPlotFlightPlanSegment(
     Planetarium const* const planetarium,
     Plugin const* const plugin,
     char const* const vessel_guid,
-    int const index) {
-  journal::Method<journal::PlanetariumPlotFlightPlanSegment> m({planetarium,
-                                                                plugin,
-                                                                vessel_guid,
-                                                                index});
+    int const index,
+    ScaledSpacePoint* const vertices,
+    int const vertices_size,
+    int* const vertex_count) {
+  journal::Method<journal::PlanetariumPlotFlightPlanSegment> m(
+      {planetarium, plugin, vessel_guid, index, vertices, vertices_size},
+      {vertex_count});
   CHECK_NOTNULL(plugin);
   CHECK_NOTNULL(planetarium);
+  *vertex_count = 0;
+
   Vessel const& vessel = *plugin->GetVessel(vessel_guid);
   CHECK(vessel.has_flight_plan()) << vessel_guid;
   auto const segment = vessel.flight_plan().GetSegment(index);
-  RP2Lines<Length, Camera> rp2_lines;
+  if (segment->empty()) {
+    return m.Return();
+  }
   // TODO(egg): this is ugly; we should centralize rendering.
   // If this is a burn and we cannot render the beginning of the burn, we
   // render none of it, otherwise we try to render the Frenet trihedron at the
   // start and we fail.
   if (index % 2 == 0 ||
-      segment->empty() ||
       segment->front().time >= plugin->renderer().GetPlottingFrame()->t_min()) {
-    rp2_lines = planetarium->PlotMethod2(*segment,
-                                         segment->begin(), segment->end(),
-                                         plugin->CurrentTime(),
-                                         /*reverse=*/false);
+    planetarium->PlotMethod3(
```

Comment:
The `PlotMethod2` that takes iterator was doing nontrivial work (cropping to the plotting frame, checking for empty trajectory).  Why is this gone?  Why don't we follow the existing pattern?

Reply:
Reverting to something resembling the existing code because of the iterators anyway.